src/lakefs_spec/util.py

```diff
@@ -25,6 +27,16 @@ def depaginate(
             break
         kwargs["after"] = resp.pagination.next_offset
 
+def ensure_path_is_str(path: Union[str, os.PathLike, pathlib.Path]) -> str:
```

Comment:
@nicholasjng I just had the idea to use a utility class instead of this one helper function to handle the paths in `lakeFS-spec` and ensure we only pass strings once it leaves our domain of concern. Let me know what you think :)

Reply:
What is the benefit of this over `pathlib.Path`? `PathHandler` is not a string, but it's not a `Path`, either.

The idea is nice, but I don't see how this is immediately useful for us. I think we should stay with fsspec's style as much as possible, which seems to be "immediately coerce to string via `stringify_path`".

So it's more function-driven instead of OOP, which makes sense for the way the interface is designed. What do you think?

(I personally think that for `(put|get)_file`, this should be as easy as calling `stringify_path` on both `lpath` and `rpath` after the precheck block. The `(put|get)` methods are another story, and might require sending a patch to `fsspec` itself, see my comment on the original issue).